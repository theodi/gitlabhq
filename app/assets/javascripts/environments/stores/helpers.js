/**
 * Deploy boards are EE only.
 *
 * @param {Object} environment
 * @returns {Object}
 */
export const setDeployBoard = (oldEnvironmentState, environment) => environment;
