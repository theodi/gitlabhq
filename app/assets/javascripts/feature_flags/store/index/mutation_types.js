export const SET_FEATURE_FLAGS_OPTIONS = 'SET_FEATURE_FLAGS_OPTIONS';

export const REQUEST_FEATURE_FLAGS = 'REQUEST_FEATURE_FLAGS';
export const RECEIVE_FEATURE_FLAGS_SUCCESS = 'RECEIVE_FEATURE_FLAGS_SUCCESS';
export const RECEIVE_FEATURE_FLAGS_ERROR = 'RECEIVE_FEATURE_FLAGS_ERROR';

export const REQUEST_USER_LISTS = 'REQUEST_USER_LISTS';
export const RECEIVE_USER_LISTS_SUCCESS = 'RECEIVE_USER_LISTS_SUCCESS';
export const RECEIVE_USER_LISTS_ERROR = 'RECEIVE_USER_LISTS_ERROR';

export const REQUEST_DELETE_USER_LIST = 'REQUEST_DELETE_USER_LIST';
export const RECEIVE_DELETE_USER_LIST_ERROR = 'RECEIVE_DELETE_USER_LIST_ERROR';

export const UPDATE_FEATURE_FLAG = 'UPDATE_FEATURE_FLAG';
export const RECEIVE_UPDATE_FEATURE_FLAG_SUCCESS = 'RECEIVE_UPDATE_FEATURE_FLAG_SUCCESS';
export const RECEIVE_UPDATE_FEATURE_FLAG_ERROR = 'RECEIVE_UPDATE_FEATURE_FLAG_ERROR';

export const REQUEST_ROTATE_INSTANCE_ID = 'REQUEST_ROTATE_INSTANCE_ID';
export const RECEIVE_ROTATE_INSTANCE_ID_SUCCESS = 'RECEIVE_ROTATE_INSTANCE_ID_SUCCESS';
export const RECEIVE_ROTATE_INSTANCE_ID_ERROR = 'RECEIVE_ROTATE_INSTANCE_ID_ERROR';

export const RECEIVE_CLEAR_ALERT = 'RECEIVE_CLEAR_ALERT';
