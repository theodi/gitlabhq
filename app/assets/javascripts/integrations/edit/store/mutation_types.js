export const SET_OVERRIDE = 'SET_OVERRIDE';
export const SET_IS_SAVING = 'SET_IS_SAVING';
export const SET_IS_TESTING = 'SET_IS_TESTING';
