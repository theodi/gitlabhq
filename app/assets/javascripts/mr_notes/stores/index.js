import Vue from 'vue';
import Vuex from 'vuex';
import batchCommentsModule from '~/batch_comments/stores/modules/batch_comments';
import notesModule from '~/notes/stores/modules';
import diffsModule from '~/diffs/store/modules';
import mrPageModule from './modules';

Vue.use(Vuex);

export const createStore = () =>
  new Vuex.Store({
    modules: {
      page: mrPageModule(),
      notes: notesModule(),
      diffs: diffsModule(),
      batchComments: batchCommentsModule(),
    },
  });

export default createStore();
