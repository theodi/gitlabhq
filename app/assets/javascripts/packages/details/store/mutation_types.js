export const SET_LOADING = 'SET_LOADING';
export const SET_PACKAGE_VERSIONS = 'SET_PACKAGE_VERSIONS';
