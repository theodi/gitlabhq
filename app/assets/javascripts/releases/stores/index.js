import Vuex from 'vuex';
import * as getters from './getters';

export default ({ modules, featureFlags }) =>
  new Vuex.Store({
    modules,
    state: { featureFlags },
    getters,
  });
