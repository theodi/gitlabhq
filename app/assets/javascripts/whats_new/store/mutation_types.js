export const CLOSE_DRAWER = 'CLOSE_DRAWER';
export const OPEN_DRAWER = 'OPEN_DRAWER';
export const SET_FEATURES = 'SET_FEATURES';
