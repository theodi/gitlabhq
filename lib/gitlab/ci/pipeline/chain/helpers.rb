# frozen_string_literal: true

module Gitlab
  module Ci
    module Pipeline
      module Chain
        module Helpers
          def error(message, config_error: false, drop_reason: nil)
            if config_error
              drop_reason = :config_error
              pipeline.yaml_errors = message
            end

            pipeline.add_error_message(message)
            pipeline.drop!(drop_reason) if drop_reason && persist_pipeline?

            # TODO: consider not to rely on AR errors directly as they can be
            # polluted with other unrelated errors (e.g. state machine)
            # https://gitlab.com/gitlab-org/gitlab/-/issues/220823
            pipeline.errors.add(:base, message)
          end

          def warning(message)
            pipeline.add_warning_message(message)
          end

          def persist_pipeline?
            command.save_incompleted && !pipeline.readonly?
          end
        end
      end
    end
  end
end
