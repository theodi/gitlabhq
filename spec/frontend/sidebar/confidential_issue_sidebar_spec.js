import { shallowMount } from '@vue/test-utils';
import { mockTracking, triggerEvent } from 'helpers/tracking_helper';
import { useMockLocationHelper } from 'helpers/mock_window_location_helper';
import ConfidentialIssueSidebar from '~/sidebar/components/confidential/confidential_issue_sidebar.vue';
import EditForm from '~/sidebar/components/confidential/edit_form.vue';
import createStore from '~/notes/stores';
import * as types from '~/notes/stores/mutation_types';

jest.mock('~/flash');
jest.mock('~/sidebar/services/sidebar_service');

describe('Confidential Issue Sidebar Block', () => {
  useMockLocationHelper();

  let wrapper;
  const mutate = jest
    .fn()
    .mockResolvedValue({ data: { issueSetConfidential: { issue: { confidential: true } } } });

  const createComponent = ({ propsData, data = {} }) => {
    const store = createStore();
    wrapper = shallowMount(ConfidentialIssueSidebar, {
      store,
      data() {
        return data;
      },
      propsData: {
        iid: '',
        fullPath: '',
        ...propsData,
      },
      mocks: {
        $apollo: {
          mutate,
        },
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  it.each`
    confidential | isEditable
    ${false}     | ${false}
    ${false}     | ${true}
    ${true}      | ${false}
    ${true}      | ${true}
  `(
    'renders for confidential = $confidential and isEditable = $isEditable',
    ({ confidential, isEditable }) => {
      createComponent({
        propsData: {
          isEditable,
        },
      });
      wrapper.vm.$store.state.noteableData.confidential = confidential;

      return wrapper.vm.$nextTick().then(() => {
        expect(wrapper.element).toMatchSnapshot();
      });
    },
  );

  describe('if editable', () => {
    beforeEach(() => {
      createComponent({
        propsData: {
          isEditable: true,
        },
      });
      wrapper.vm.$store.state.noteableData.confidential = true;
    });

    it('displays the edit form when editable', () => {
      wrapper.setData({ edit: false });

      return wrapper.vm
        .$nextTick()
        .then(() => {
          wrapper.find({ ref: 'editLink' }).trigger('click');
          return wrapper.vm.$nextTick();
        })
        .then(() => {
          expect(wrapper.find(EditForm).exists()).toBe(true);
        });
    });

    it('displays the edit form when opened from collapsed state', () => {
      wrapper.setData({ edit: false });

      return wrapper.vm
        .$nextTick()
        .then(() => {
          wrapper.find({ ref: 'collapseIcon' }).trigger('click');
          return wrapper.vm.$nextTick();
        })
        .then(() => {
          expect(wrapper.find(EditForm).exists()).toBe(true);
        });
    });

    it('tracks the event when "Edit" is clicked', () => {
      const spy = mockTracking('_category_', wrapper.element, jest.spyOn);

      const editLink = wrapper.find({ ref: 'editLink' });
      triggerEvent(editLink.element);

      expect(spy).toHaveBeenCalledWith('_category_', 'click_edit_button', {
        label: 'right_sidebar',
        property: 'confidentiality',
      });
    });
  });
  describe('computed confidential', () => {
    beforeEach(() => {
      createComponent({
        propsData: {
          isEditable: true,
        },
      });
    });

    it('returns false when noteableData is not present', () => {
      wrapper.vm.$store.commit(types.SET_NOTEABLE_DATA, null);

      expect(wrapper.vm.confidential).toBe(false);
    });

    it('returns true when noteableData has confidential attr as true', () => {
      wrapper.vm.$store.commit(types.SET_NOTEABLE_DATA, {});
      wrapper.vm.$store.commit(types.SET_ISSUE_CONFIDENTIAL, true);

      expect(wrapper.vm.confidential).toBe(true);
    });

    it('returns false when noteableData has confidential attr as false', () => {
      wrapper.vm.$store.commit(types.SET_NOTEABLE_DATA, {});
      wrapper.vm.$store.commit(types.SET_ISSUE_CONFIDENTIAL, false);

      expect(wrapper.vm.confidential).toBe(false);
    });

    it('returns true when confidential attr is true', () => {
      wrapper.vm.$store.commit(types.SET_NOTEABLE_DATA, {});
      wrapper.vm.$store.commit(types.SET_ISSUE_CONFIDENTIAL, true);

      expect(wrapper.vm.confidential).toBe(true);
    });

    it('returns false when confidential attr is false', () => {
      wrapper.vm.$store.commit(types.SET_NOTEABLE_DATA, {});
      wrapper.vm.$store.commit(types.SET_ISSUE_CONFIDENTIAL, false);

      expect(wrapper.vm.confidential).toBe(false);
    });
  });
});
