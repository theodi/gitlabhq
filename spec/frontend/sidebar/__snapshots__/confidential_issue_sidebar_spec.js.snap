// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confidential Issue Sidebar Block renders for confidential = false and isEditable = false 1`] = `
<div
  class="block issuable-sidebar-item confidentiality"
  iid=""
>
  <div
    class="sidebar-collapsed-icon"
    title="Not confidential"
  >
    <gl-icon-stub
      aria-hidden="true"
      name="eye"
      size="16"
    />
  </div>
   
  <div
    class="title hide-collapsed"
  >
    
    Confidentiality
    
    <!---->
  </div>
   
  <div
    class="value sidebar-item-value hide-collapsed"
  >
    <!---->
     
    <div
      class="no-value sidebar-item-value"
      data-testid="not-confidential"
    >
      <gl-icon-stub
        aria-hidden="true"
        class="sidebar-item-icon inline"
        name="eye"
        size="16"
      />
      
      Not confidential
    
    </div>
  </div>
</div>
`;

exports[`Confidential Issue Sidebar Block renders for confidential = false and isEditable = true 1`] = `
<div
  class="block issuable-sidebar-item confidentiality"
  iid=""
>
  <div
    class="sidebar-collapsed-icon"
    title="Not confidential"
  >
    <gl-icon-stub
      aria-hidden="true"
      name="eye"
      size="16"
    />
  </div>
   
  <div
    class="title hide-collapsed"
  >
    
    Confidentiality
    
    <a
      class="float-right confidential-edit"
      data-track-event="click_edit_button"
      data-track-label="right_sidebar"
      data-track-property="confidentiality"
      href="#"
    >
      Edit
    </a>
  </div>
   
  <div
    class="value sidebar-item-value hide-collapsed"
  >
    <!---->
     
    <div
      class="no-value sidebar-item-value"
      data-testid="not-confidential"
    >
      <gl-icon-stub
        aria-hidden="true"
        class="sidebar-item-icon inline"
        name="eye"
        size="16"
      />
      
      Not confidential
    
    </div>
  </div>
</div>
`;

exports[`Confidential Issue Sidebar Block renders for confidential = true and isEditable = false 1`] = `
<div
  class="block issuable-sidebar-item confidentiality"
  iid=""
>
  <div
    class="sidebar-collapsed-icon"
    title="Confidential"
  >
    <gl-icon-stub
      aria-hidden="true"
      name="eye-slash"
      size="16"
    />
  </div>
   
  <div
    class="title hide-collapsed"
  >
    
    Confidentiality
    
    <!---->
  </div>
   
  <div
    class="value sidebar-item-value hide-collapsed"
  >
    <!---->
     
    <div
      class="value sidebar-item-value hide-collapsed"
    >
      <gl-icon-stub
        aria-hidden="true"
        class="sidebar-item-icon inline is-active"
        name="eye-slash"
        size="16"
      />
      
      This issue is confidential
    
    </div>
  </div>
</div>
`;

exports[`Confidential Issue Sidebar Block renders for confidential = true and isEditable = true 1`] = `
<div
  class="block issuable-sidebar-item confidentiality"
  iid=""
>
  <div
    class="sidebar-collapsed-icon"
    title="Confidential"
  >
    <gl-icon-stub
      aria-hidden="true"
      name="eye-slash"
      size="16"
    />
  </div>
   
  <div
    class="title hide-collapsed"
  >
    
    Confidentiality
    
    <a
      class="float-right confidential-edit"
      data-track-event="click_edit_button"
      data-track-label="right_sidebar"
      data-track-property="confidentiality"
      href="#"
    >
      Edit
    </a>
  </div>
   
  <div
    class="value sidebar-item-value hide-collapsed"
  >
    <!---->
     
    <div
      class="value sidebar-item-value hide-collapsed"
    >
      <gl-icon-stub
        aria-hidden="true"
        class="sidebar-item-icon inline is-active"
        name="eye-slash"
        size="16"
      />
      
      This issue is confidential
    
    </div>
  </div>
</div>
`;
